examples/unpack-archive.rs

```diff
@@ -0,0 +1,19 @@
+use std::{fs, io::Read, path::PathBuf};
+use tar::Archive;
+use xz2::read::XzDecoder;
+
+fn unpack_tar_xz(archive_path: &PathBuf, dest: &PathBuf) {
+    let archive_bytes = fs::read(archive_path).expect("Problem reading archive as bytes");
+
+    let mut tar: Vec<u8> = Vec::new();
+    let mut decompressor = XzDecoder::new(&archive_bytes[..]);
+    decompressor.read_to_end(&mut tar).expect("Problem decompressing archive");
+
+    // We've decompressed the .xz; now unpack the tar.
+    let mut archive = Archive::new(&tar[..]);
```

Comment:
I don't think that the decompress/extract step has to happen in two phases, the `Archive` could be passed `XzDecoder` directly, and additionally we don't need `fs::read` but rather the `XzDecoder` could be passed the raw file.

Reply:
Just stumbled upon this and wondering if [here](https://github.com/rust-lang/cargo/blob/3cd05e28950f0fe4e1d14bc14162c7e2661af397/src/cargo/ops/cargo_package.rs#L631-L642) the `decompress/extract step has to happen in two phases` because there's no other way to `archive.set_preserve_mtime(false);` if it were to happen in one phase ?

Reply:
I think you can just call that on `archive` at any time after construction, and constructing an `Archive` doesn't actually do any decompression until it's needed to do so.